core/shared/src/main/scala/spire/math/package.scala

```diff
@@ -403,10 +403,126 @@ package object math {
   final def hypot[@sp(Float, Double) A](x: A, y: A)
     (implicit f: Field[A], n: NRoot[A], o: Order[A]): A = {
     import spire.implicits._
-    if (x > y) x.abs * (1 + (y/x)**2).sqrt
-    else y.abs * (1 + (x/y)**2).sqrt
+    def abs(n: A): A = if (n < f.zero) -n else n
+    if (x > y) abs(x) * (1 + (y/x)**2).sqrt
+    else abs(y) * (1 + (x/y)**2).sqrt
   }
 
+  // BigInt
+  /**
+   * This will return the largest integer that meets some criteria. Specifically,
+   * if we're looking for some integer `x` and `f(x')` is guaranteed to return
+   * `true` iff `x' <= x`, then this will return `x`.
+   *
+   * This can be used, for example, to find an integer `x` s.t. 
+   * `x * x < y < (x+1)*(x+1)`, by using `intSearch(x => x * x <= y)`.
+   */
+  private def intSearch(f: Int => Boolean): Int = {
```

Comment:
I only see this as an addition, without the corresponding existing code deleted. Does either this or the old version need to be removed?